pkg/authorization/workspace_content_authorizer.go

```diff
@@ -100,91 +101,89 @@ func (a *workspaceContentAuthorizer) Authorize(ctx context.Context, attr authori
 	)
 
-	// TODO: decide if we want to require workspaces for all kcp variations. For now, only check if the workspace controllers are running,
-	// as that ensures the ClusterWorkspace CRD is installed, and that our shared informer factory can sync all its caches successfully.
-	if a.clusterWorkspaceLister != nil {
-		// check the workspace even exists
-		// TODO: using scoping when available
-		if ws, err := a.clusterWorkspaceLister.Get(clusters.ToClusterAwareKey(parentClusterName, clusterWorkspace)); err != nil {
-			if errors.IsNotFound(err) {
-				return authorizer.DecisionDeny, fmt.Sprintf("%q workspace access not permitted", cluster.Name), nil
-			}
-			return authorizer.DecisionNoOpinion, "", err
-		} else if len(ws.Status.Initializers) > 0 {
+	extraGroups := sets.NewString()
+
+	// check the workspace even exists
+	if ws, err := a.clusterWorkspaceLister.Get(clusters.ToClusterAwareKey(parentClusterName, clusterWorkspace)); err != nil {
+		if errors.IsNotFound(err) {
+			return authorizer.DecisionDeny, fmt.Sprintf("%q workspace access not permitted", cluster.Name), nil
+		}
+		return authorizer.DecisionNoOpinion, "", err
+	} else if len(ws.Status.Initializers) > 0 {
+		// service accounts can potentially access sub-workspaces, but not other workspaces during initialization. Hence, these two case:
+		// 1. it's  not a serivce account
+		// 2. it's a service account, and it is defined in the parent workspace
+		if len(subjectClusters) == 0 && subjectClusters[parentClusterName] {
 			workspaceAttr := authorizer.AttributesRecord{
 				User:            attr.GetUser(),
-				Verb:            attr.GetVerb(),
+				Verb:            "initialize",
 				APIGroup:        v1alpha1.SchemeGroupVersion.Group,
 				APIVersion:      v1alpha1.SchemeGroupVersion.Version,
 				Resource:        "clusterworkspaces",
-				Subresource:     "initialize",
+				Subresource:     "content",
 				Name:            clusterWorkspace,
 				ResourceRequest: true,
 			}
-
 			dec, reason, err := parentAuthorizer.Authorize(ctx, workspaceAttr)
 			if err != nil {
 				return dec, reason, err
 			}
-			if dec != authorizer.DecisionAllow {
-				return dec, fmt.Sprintf("%q workspace access not permitted", cluster.Name), nil
-			}
-		}
-	}
-
-	extraGroups := []string{}
-	if subjectCluster := attr.GetUser().GetExtra()[authserviceaccount.ClusterNameKey]; len(subjectCluster) > 0 {
-		// a subject from a workspace, like a ServiceAccount, is automatically authenticated
-		// against that workspace.
-		// On the other hand, referencing that in the parent cluster for further permissions
-		// is not possible. Hence, we skip the authorization steps for the verb below.
-		for _, sc := range subjectCluster {
-			if logicalcluster.New(sc) == cluster.Name {
-				extraGroups = append(extraGroups, bootstrap.SystemKcpClusterWorkspaceAccessGroup)
-				break
+			if dec == authorizer.DecisionAllow {
+				extraGroups.Insert(bootstrap.SystemKcpClusterWorkspaceAccessGroup)
```

Comment:
additionally, add `SystemKcpClusterWorkspaceAdminGroup`. Reasoning: if the SA has `initialize` permissions on the parent ws, it is expected that this SA has administrative permissions on sub-workspaces that have initializers declared.

The initializing controller (the one who has a SA on the parent ws with `initialize` verbs) thus has the same permissions as an `admin` user manually initializing a universal sub-workspace.